### PR TITLE
[Google_Workspace] Moving edge processing to ingest pipeline

### LINKED ISSUE #659: Convert Google Workspace's edge processing to Ingest Node pipeline
# Convert edge processing to Ingest Node pipeline

This package uses Beats processors for some of its data processing. We want to
move that processing into the ingest node pipeline that is part of the package
to make reuse easier (e.g. data from Kafka could be routed through the pipeline).

### Data Streams
- admin (inputs: httpjson, log)
  - [x] All beats processors are removed
  - [x] [Pipeline tests](https://github.com/elastic/elastic-package/blob/master/docs/howto/pipeline_testing.md) added
- drive (inputs: httpjson, log)
  - [x] All beats processors are removed
  - [x] [Pipeline tests](https://github.com/elastic/elastic-package/blob/master/docs/howto/pipeline_testing.md) added
- groups (inputs: httpjson, log)
  - [x] All beats processors are removed
  - [x] [Pipeline tests](https://github.com/elastic/elastic-package/blob/master/docs/howto/pipeline_testing.md) added
- login (inputs: httpjson, log)
  - [x] All beats processors are removed
  - [x] [Pipeline tests](https://github.com/elastic/elastic-package/blob/master/docs/howto/pipeline_testing.md) added
- saml (inputs: httpjson, log)
  - [x] All beats processors are removed
  - [x] [Pipeline tests](https://github.com/elastic/elastic-package/blob/master/docs/howto/pipeline_testing.md) added
- user_accounts (inputs: httpjson, log)
  - [x] All beats processors are removed
  - [x] [Pipeline tests](https://github.com/elastic/elastic-package/blob/master/docs/howto/pipeline_testing.md) added

### PR DESCRIPTION
**Pull request by P1llus**
## What does this PR do?

This moves all edge processing to ingest pipelines for the google_workspace package. It also adds some new testdata and pipeline tests.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/master/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.

## Related issues

- Closes #659

### CODE REVIEW

**Comment by P1llus** [comment]
Work in progress, adding each fileset once they are completed.

File: packages/google_workspace/data_stream/admin/elasticsearch/ingest_pipeline/default.yml [hunk 1]
```diff
@@ -1,6 +1,716 @@
...
...ER_IM","ENABLE_USER_IP_WHITELIST","CHANGE_USER_KEYWORD","CHANGE_USER_LANGUAGE","CHANGE_USER_LOCATION","CHANGE_USER_ORGANIZATION","CHANGE_USER_PHONE_NUMBER","CHANGE_RECOVERY_EMAIL","CHANGE_RECOVERY_PHONE","CHANGE_USER_RELATION","CHANGE_USER_ADDRESS","GRANT_DELEGATED_ADMIN_PRIVILEGES","CHANGE_FIRST_NAME","GMAIL_RESET_USER","CHANGE_LAST_NAME","MAIL_ROUTING_DESTINATION_ADDED","MAIL_ROUTING_DESTINATION_REMOVED","ADD_NICKNAME","REMOVE_NICKNAME","CHANGE_PASSWORD","CHANGE_PASSWORD_ON_NEXT_LOGIN","REMOVE_RECOVERY_EMAIL","REMOVE_RECOVERY_PHONE","RESET_SIGNIN_COOKIES","SECURITY_KEY_REGISTERED_FOR_USER","REVOKE_SECURITY_KEY","TURN_OFF_2_STEP_VERIFICATION","UNBLOCK_USER_SESSION","UNENROLL_USER_FROM_TITANIUM","ARCHIVE_USER","UPDATE_BIRTHDATE","DOWNGRADE_USER_FROM_GPLUS","USER_ENROLLED_IN_TWO_STEP_VERIFICATION","MOVE_USER_TO_ORG_UNIT","USER_PUT_IN_TWO_STEP_VERIFICATION_GRACE_PERIOD","RENAME_USER","UNENROLL_USER_FROM_STRONG_AUTH","SUSPEND_USER","UNARCHIVE_USER","UNSUSPEND_USER","UPGRADE_USER_TO_GPLUS","MOBILE_DEVICE_APPROVE","MOBILE_DEVICE_BLOCK","MOBILE_DEVICE_WIPE","MOBILE_ACCOUNT_WIPE","MOBILE_DEVICE_CANCEL_WIPE_THEN_APPROVE","MOBILE_DEVICE_CANCEL_WIPE_THEN_BLOCK"].contains(ctx?.event?.action)'
```
Comment (marc-gr) [inline packages/google_workspace/data_stream/admin/elasticsearch/ingest_pipeline/default.yml:89]: only if you want and have the time, maybe the categorization in the pipelines could be simplified by a parameterized script processor instead, eg:

```
  - script:
      description: Set event category and type for all event types.
      lang: painless
      params:
        "CHANGE_APPLICATION_SETTING":
          category:
            - configuration
          type:
            - change
      if: ctx?.event?.action != null
      source: |-
        def hm = new HashMap(params[ctx.event.action]);
        hm.forEach((k, v) -> ctx.event[k] = v);
```

I also thinks makes for a better readability

File: packages/google_workspace/data_stream/drive/elasticsearch/ingest_pipeline/default.yml [hunk 2]
```diff
@@ -1,6 +1,234 @@
...
+      field: event.type
+      value: deletion
+      if: '["delete", "trash"].contains(ctx?.event?.action)'
+  - append: 
+      field: event.type
+      value: info
+      if: '["download", "preview", "print", "view"].contains(ctx?.event?.action)'
+  - script:
+      lang: painless
+      if: 'ctx?.json?.events?.parameters != null && ctx?.json?.events?.parameters instanceof List'
```
Comment (marc-gr) [inline packages/google_workspace/data_stream/drive/elasticsearch/ingest_pipeline/default.yml:125]: there are some of this script processors where maybe a description could be useful to know what they do

**Review by marc-gr — APPROVED** [review]
LGTM, left a couple comments that would be nice to have IMO

**Comment by P1llus** [comment]
After some discussion, this will be merged for now, as it only runs once every 2 hours, also any additional things we might want to add can be done later, as this is a blocking change.

**Pull request merged**